docs/mcp/config.md

```diff
@@ -133,12 +133,108 @@ For tighter security control, you can configure the server to impersonate a spec
+        url: oci://ghcr.io/controlplaneio-fluxcd/flux-operator-manifests
+        ref:
+          tag: << inputs.version >>
+    - apiVersion: kustomize.toolkit.fluxcd.io/v1
+      kind: Kustomization
+      metadata:
+        name: flux-operator-mcp
+        namespace: flux-system
+      spec:
+        serviceAccountName: flux-operator
+        interval: 60m
+        wait: true
+        timeout: 5m
+        retryInterval: 5m
+        prune: true
+        sourceRef:
+          kind: OCIRepository
+          name: flux-operator-manifests
+        path: ./flux-operator-mcp
+        patches:
+          - patch: |
+              - op: add
+                path: /spec/template/spec/containers/0/args/-
+                value: --read-only=<< inputs.readonly >>
+            target:
+              kind: Deployment
+    - kind: NetworkPolicy
+      apiVersion: networking.k8s.io/v1
+      metadata:
+        name: flux-operator-mcp
+        namespace: flux-system
+      spec:
+        policyTypes:
+          - Ingress
+        podSelector:
+          matchLabels:
+            app.kubernetes.io/name: flux-operator-mcp
+        ingress:
+          - from:
+              - namespaceSelector:
+                  matchLabels:
+                    kubernetes.io/metadata.name: << inputs.accessFrom >>
+            ports:
+              - protocol: TCP
+                port: 9090
+```
+
+This ResourceSet will create a Kubernetes Deployment for the Flux MCP Server
+with cluster-admin permissions. It is recommended to set the `readonly` input to `true`
+in production environments to prevent modifications to the cluster state.
+
+The server is exposed via a Kubernetes Service named `flux-operator-mcp`
+in the `flux-system` namespace, listening on port `9090`. If the MCP client
+is running in-cluster, the `accessFrom` input should be set to the name of the
+namespace where the MCP client is deployed.
 
-While it is possible to run the Flux MCP Server in a Docker container and mount the kubeconfig file,
-this could work only when the kubeconfig contains static credentials. If you connect to a
-managed Kubernetes cluster, the authentication implies shelling out to the cloud provider CLI
-with access to the other config files.
+To connect to the server, start port forwarding with:
+
+```shell
+kubectl port-forward -n flux-system svc/flux-operator-mcp 9090:9090
+```
+
+Then, in your VS Code settings, add:
+
+```json
+{
+ "mcp": {
+   "servers": {
+     "flux-operator-mcp": {
+       "type": "sse",
+       "url": "http://localhost:9090/sse"
+     }
+   }
+ }
+}
+```
+
+!!! warning "Warning"
 
-At the moment, we do not provide a Docker image for the Flux MCP Server, but you can build one
-using the binaries available on the [release page](https://github.com/controlplaneio-fluxcd/flux-operator/releases).
+    Note that when running in-cluster, the kubeconfig context switching tools are disabled,
+    so comparing deployments across clusters is not possible.
```

Comment:
It should be possible to use a kubeconfig in-cluster too, of course not one with exec plugins etc., but in every cluster you can create a KSA token and make a kubeconfig with it, and you can have one with multiple clusters. In our code right now we don't allow that due to `inCluster := os.Getenv("KUBERNETES_SERVICE_HOST") != ""`, but we could look first for `KUBECONFIG` and if not set fallback to in-cluster. Of course, static KSA tokens suck, so maybe we shouldn't allow that.

One thing we can definitely support though, is upcoming workload identity support for managed services :) The MCP server SA can be granted access on remote EKS/AKS/GKE clusters :)